presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java

```diff
@@ -804,11 +814,153 @@ public void testSchemaEvolution()
         assertUpdate("DROP TABLE test_schema_evolution");
     }
 
+    @Test
+    public void testRcAndTextFormats()
+            throws IOException
+    {
+        getQueryRunner().execute("CREATE TABLE lineitem_ex_partitioned WITH (format='ORC', partitioned_by=ARRAY['ds']) AS " +
+                "SELECT linenumber, orderkey, partkey, suppkey, quantity, extendedprice, tax, shipinstruct, shipmode, ship_by_air, " +
+                "is_returned, ship_day, ship_month, ship_timestamp, commit_timestamp, discount_real, discount, tax_real, ship_day_month, " +
+                "discount_long_decimal, tax_short_decimal, long_decimals, keys, doubles, nested_keys, flags, reals, info, dates, timestamps, " +
+                "comment, uppercase_comment, fixed_comment, char_array, varchar_array, '2019-11-01' AS ds FROM lineitem_ex");
+        try {
+            for (HiveStorageFormat format : ImmutableList.of(RCBINARY, RCTEXT, TEXTFILE)) {
+                assertFileFormat(format);
+            }
+        }
+        finally {
+            assertUpdate("DROP TABLE lineitem_ex_partitioned");
+        }
+    }
+
+    private void assertFileFormat(HiveStorageFormat storageFormat)
+            throws IOException
+    {
+        QueryRunner queryRunner = getQueryRunner();
+        try {
+            queryRunner.execute("CREATE TABLE test_file_format_orc WITH (format='ORC', partitioned_by=ARRAY['ds']) AS " +
+                    "SELECT * FROM lineitem_ex_partitioned LIMIT 1");
+
+            queryRunner.execute(format("CREATE TABLE test_file_format WITH (format='%s', partitioned_by=ARRAY['ds']) AS " +
+                    "SELECT * FROM lineitem_ex_partitioned", storageFormat));
+
+            Path orcDirectory = getPartitionDirectory("test_file_format_orc", "ds='2019-11-01'");
+            deleteRecursively(orcDirectory, ALLOW_INSECURE);
+
+            Path otherDirectory = getPartitionDirectory("test_file_format", "ds='2019-11-01'");
+            Files.move(otherDirectory, orcDirectory, REPLACE_EXISTING);
+
+            String cte = "WITH test_file_format_orc AS (\n" +
```

Comment:
Let's not copy-paste the definition of lineitem_ex table and re-use WITH_LINEITEM_EX instead:

```
            String cte = WITH_LINEITEM_EX + ", test_file_format_orc AS " +
                    "(SELECT\n" +
                    "    linenumber,\n" +
                    "    orderkey,\n" +
                    "    partkey,\n" +
                    "    suppkey,\n" +
                    "    quantity,\n" +
                    "    extendedprice,\n" +
                    "    tax,\n" +
                    "    shipinstruct,\n" +
                    "    shipmode,\n" +
                    "    ship_by_air,\n" +
                    "    is_returned,\n" +
                    "    ship_day,\n" +
                    "    ship_month,\n" +
                    "    ship_timestamp,\n" +
                    "    commit_timestamp,\n" +
                    "    discount_real,\n" +
                    "    discount,\n" +
                    "    tax_real,\n" +
                    "    ship_day_month,\n" +
                    "    discount_long_decimal,\n" +
                    "    tax_short_decimal,\n" +
                    "    long_decimals,\n" +
                    "    keys,\n" +
                    "    doubles,\n" +
                    "    nested_keys,\n" +
                    "    flags,\n" +
                    "    reals,\n" +
                    "    info,\n" +
                    "    dates,\n" +
                    "    timestamps,\n" +
                    "    comment,\n" +
                    "    uppercase_comment,\n" +
                    "    fixed_comment,\n" +
                    "    char_array,\n" +
                    "    varchar_array,\n" +
                    "    '2019-11-01' AS ds\n" +
                    "FROM lineitem_ex)";
```